gestalt/vault.py

```diff
@@ -0,0 +1,69 @@
+import requests
+from typing import Optional
+import hvac
+
+
+class Vault():
```

Comment:
Wouldn't we want to inherit from the abstract Class `Provider`

Reply:
Circular dependency :(

Reply:
Fixed the problem of circular dependency